streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -358,6 +363,28 @@ public void pipeInput(final ConsumerRecord<byte[], byte[]> consumerRecord) {
         }
     }
 
+    private void validateSourceTopicNameRegexPattern(String consumerTopic) {
+        for (final String sourceTopicName : internalTopologyBuilder.getSourceTopicNames()) {
+            if (!sourceTopicName.equals(consumerTopic) && Pattern.compile(sourceTopicName).matcher(consumerTopic).matches()) {
```

Comment:
Thinking about this twice, and considering you explanation what we can do and what not, I am now wondering if this checks adds a lot of value.

\cc @bbejeck @vvcephei @guozhangwang